docs/nativeaot.md

```diff
@@ -0,0 +1,49 @@
+# NativeAOT
+
+We've added *experimental* support for using [NativeAOT][1] when publishing iOS,
+tvOS, Mac Catalyst and macOS apps in .NET 8.
+
+NativeAOT may produce smaller and/or faster apps - or it may not. It's very
+important test and profile to determine the results of enabling NativeAOT.
+
+NativeAOT is enabled by setting the `PublishAot` property to `true` in the project file:
+
+```xml
+<PropertyGroup>
+	<PublishAot>true</PublishAot>
+</PropertyGroup>
+```
+
+## Notes
+
+NativeAOT is only used when publishing (`dotnet publish`). In particular
+`dotnet build -t:Publish` is _not_ equivalent to `dotnet publish`, only
+`dotnet publish` will enable NativeAOT.
+
+**Unsupported** workaround: set the `_IsPublishing` property to `true` to make
+`dotnet build` think it's `dotnet publish`:
+
+```xml
+<PropertyGroup>
+	<PublishAot>true</PublishAot>
+	<_IsPublishing>true</_IsPublishing>
+</PropertyGroup>
+```
+
+This can be useful to install and run apps with `NativeAOT` from the IDE,
+because it's not possible to use `dotnet publish` when running from the IDE.
+
+Note that NativeAOT does not support managed debugging.
+
+There's no interpreter when using NativeAOT, and as such the `UseInterpreter`
+and `MtouchInterpreter` properties have no effect.
+
+NativeAOT requires trimming, and `MAUI` isn't trimmer-safe, and thus
+unfortunately `MAUI` projects don't typically work with NativeAOT (we hope to
+rectify this situation for .NET 9).
+
+There are no known issues specific to our platforms with NativeAOT; but the
+[limitations][2] are exactly the same as for other platforms.
```

Comment:
```suggestion
There are no known issues specific to our platforms with NativeAOT; but the
[limitations][2] are exactly the same as for other supported platforms. 
Nevertheless, we would like to point out a few features that are not available with NativeAOT, that are with Mono, when targeting Apple platforms:

- NativeAOT does not support managed debugging.
- There's no interpreter when using NativeAOT, and as such the `UseInterpreter`
and `MtouchInterpreter` properties have no effect.
- NativeAOT requires trimming, and `MAUI` isn't trimmer-safe, and thus
unfortunately `MAUI` projects don't typically work with NativeAOT (we hope to
rectify this situation for .NET 9).
```

Reply:
@rolfbjarne regarding the issue with MAUI not being trimmer-safe, do we want to mention that setting `TrimMode=partial` could help?

On that note, should we maybe generalise the problem to all non-trimmable assemblies?

Reply:
I never tested MAUI projects, but I was under the impression they didn't work, no workaround possible?

Reply:
I will take a look, I believe I had an example where `TrimMode=partial` helped, and still produced smaller app than Mono - I will double-check.
I just felt that `don't typically work` is a bit harsh (even though its true).

Reply:
Here is the gist with analysis of building/running 5 different MAUI iOS apps with NativeAOT using different trimming modes: https://gist.github.com/ivanpovazan/4756bdb65a34366f0944e99f601379f3 

I did not go deeper with examining why something failed, but just wanted to give an overview regarding `TrimMode=partial` vs `TrimMode=full`